### ISSUE TITLE
[EN] [NPC] [Quest] Lorna Crowley crash

### ISSUE THREAD

**Issue by galhast** [Atlantiss/BugTracker#2493]
When I arrived at the Crowley orchard, the quest marker pointed to a spot outside her house where no NPC was standing. After a while, she appeared to spawn in, and I received the horse gathering quest. I returned one horse to her and she was visible. But after that, she vanished again, and even though the horses would return to her properly, she didn't become visible again. When the quest was completed, I saw the golden ? floating where she should've been standing, but I could not interact with her.

At this point, I logged out, to get her to respawn. Now whenever I log into this character, I crash to the desktop with error #132. I can use my other characters fine.

Error log: http://pastebin.com/fThnMt6F

**Comment by xmesaj2** [comment]
Don't forget to provide IDs next time 
http://www.wowhead.com/npc=43727

> When the quest was completed, I saw the golden ? floating where she should've been standing, but I could not interact with her.

NPC has few phases during quest chains. In above link you can see it's changing location for quest states (probably Quest discovered message should show up, but it's not a rule)

**Comment by galhast** [comment]
Thanks for posting it for me. Perhaps I'm stuck outside of a phase?

**Comment by Tj20** [comment]
Just to clarify: it's not 43727 Lorna Crowley, but this one:
http://www.wowhead.com/npc=36457/lorna-crowley

**Comment by Tj20** [comment]
I managed to reproduce this bug once, but I'm not sure when exactly it does happen and when not. Still, in the meantime some pictures:

**1. Deleted my Cache, went onto the PTR and tried the Worgen area first with a new level 1 hunter.** Ok, being the PTR he was 85 already. Btw, I'm using the "official" Atlantiss mini client.

Quest "As the land shatters" was fine.
![pic](http://i.imgur.com/AkBU1kP.jpg)

Some minutes later: Lorna Crowley is there, everything works fine.
![pic](http://i.imgur.com/YfCO7cQ.jpg)

Could bring all horses and end the quest without problems.The rest of the starting area also was fine.
![pic](http://i.imgur.com/UuDig8C.jpg)

**2. Now a new level 1 char on the live realm without deleting the cache.**

Drowning Watchman are in a different position.
![pic](http://i.imgur.com/f1ky6XG.jpg)

Lorna Crowley offers me the quest as before.
![pic](http://i.imgur.com/KAGoBE8.jpg)

Entering the horse area: Strange, suddenly the ambient sounds died, total silence. I didn't notice that before (but didn't specifically look for it either).

![pic](http://i.imgur.com/5C5xvHu.jpg)

Could return the first horse but Lorna Crowley vanished. (Btw there were no other players nearby, the combat log was empty)

![pic](http://i.imgur.com/6vOtX1A.jpg)

I went back north to Duskhaven and found that area was fine, the NPCs in the village were still there like I left them. I wanted to go back to the Crowley Orchard, however on the zone change Duskhaven -> Gilneas the game crashed.

![pic](http://i.imgur.com/MYS8XTE.jpg)

Notable differences:
- I did not clear my cache, and I had done the zone with another Worgen before. Maybe that messed things up?
- First time was alone on the PTR as a level 85 player, 2nd time with about 5 other players in Gilneas on live as a low level char. Nobody was in the immediate vicinity however (except right when I crashed somebody was fighting Lucius the Cruel)
- Different class: druid instead of hunter. I accepted the quest in cat form and returned the first horse in Worgen form
- this time I had accepted a random guild invitation and was in a guild
- this time in Duskhaven I logged out and back in again to send me some gold

**Comment by Tj20** [comment]
Part II, or: *how I permanently killed my character*

Using the unstuck option of the shop I'm back in Gilneas ([pic](http://i.imgur.com/SDnVNf8.jpg)). Thank god I had put my new hearthstone location to Duskhaven, otherwise I would be totally f***ed now: There's now way from Gilneas City to Duskhaven. 

![you shall not pass, unless your Hearthstone is in Duskhaven](http://i.imgur.com/J6Kmmrs.jpg)

Since my hearthstone was in Duskhaven I could port back. Great success: Lorna Crowley is there and I can continue my quest! This time I set her as my focus target, give her a raid marker and ride the next horse home.

![pic](http://i.imgur.com/OnfSt4s.jpg)

**Now we're on to something: When I run back to horse number 3, I switch to cat form. As soon as my zone changes to the horse area again, my focus target vanishes. When I return, Lorna is lost again. I'm beginning to suspect that either I can't use cat form while changing zones, or I was too far away from Lorna.**

![pic](http://i.imgur.com/LH6g9YI.jpg)

I log out and log in again. Lorna ist still away and the game crashes at login.

![pic](http://i.imgur.com/zJwvH2u.jpg)

I use the unstuck option a second time.

**Now I'm having a brilliant idea: "Let me delete my cache folder and abandon the quest, that will surely fix it!" I l log back into Gilneas City, use my hearthstone to Duskhaven, and as soon as I get past the border Duskhaven -> Gilneas... the game crashes again. Damn.**

![pic](http://i.imgur.com/e2wFP4c.jpg)

I'm using unstuck -> hearthstone to Duskhaven a 3rd time. This time I reach Crowley's Orchard, but Lorna is not there. 

![pic](http://i.imgur.com/kO0vCvT.jpg)

After a relog my game crashes again. I've investested 15 vote points by now and decide to call it a day. This character is f***ed, I better should delete it and start a new one.

**Bottom line:** I think Lorna disappears either when you enter the horse area as a druid in cat form, or when you're too far away from her. Because more tests seem to be impossible with this character (I think because I deleted my cache folder and anbandoned the quest and now I can't get into the correct phase again) I decided to call it a day though.

Another problem, as mentioned before: if you didn't set your hearthstone to Duskhaven your character is also lost. The unstuck option puts you back to Gilneas City but you can't leave the city because the way to Duskhaven is closed.

**Comment by Wolffenstein** [comment]
Thank you for testing, I'll surely look into it.

**Comment by Tj20** [comment]
Thanks! 

(changed some direct image links into text links... apparently github doesn't like too many images on one page)

**Comment by Tj20** [comment]
Ok, today I've made a new char and tested this a bit more.

One thing seems certain: This quest is a Bermuda triangle at the moment. 

When Lorna disappeared for me I sat down and waited a bit to see if she would reappear after some time. She did not, but I saw a couple of people appearing and disappearing. I've put them on my friendlist and it looked like this:
![pic](http://i.imgur.com/8fx4Jp9.jpg)

[pic 2](http://i.imgur.com/tJRjgmr.jpg)

- 2 people completed the quest and moved on
- 3 people obviously had client crashes: the went online and offline. Finally they stopped. Then my client also crashed.

**2 out of 6 people are able to complete this quest and can proceed to play their Worgen... not a good quota....**

Regarding the problem itself:

My previous theories were wrong. This time I deleted my cache before making a new Worgen on a different account, and it wasn't a druid but a hunter. Still Lorna disappeared after some testing.

**Now I'm suspecting that it's either a specific horse or a location near a horse which makes Lorna disappear (less likely) or that other people are on the same quest as me, and then somehow the phasing gets messed up.** This might also explain why I had no problems on the PTR: there were simple no other players which could disturb me doing the quest.

![pic](http://i.imgur.com/oalS1IF.jpg)

When standing near these two horses (they're so far away that Lorna is just out of render distance) I noticed the problem began (Lorna disappeared) but also after some back and forth Lorna appeared again. [Koroth the Hillbreaker](http://www.wowhead.com/npc=36528/koroth-the-hillbreaker#comments)'s path runs through them, maybe he has something to do with it?

I didn't have to mount the horse, I just ran back and forth a bit. Sometimes Lorna was there, sometimes she was not.

When Lorna was away, I also ran back to Duskhaven until she was out of sight (37,67). When leaving zone Duskhaven and entering zone Gilneas (37,68) Lorna was back in front of her house again, [pic](http://i.imgur.com/WqmCGDD.jpg)

**Bottom line:** I think when somebody else is near Lorna and you leave the area around her, she disappears. 
Do not log out when she's away, this will crash your client! Instead, run back to Duskhaven or back to the far end of the horse area. If nobody else is near Lorna, you will see her again and you can turn in the quest. At least that seems the most likely explanation to me.

**Comment by Tj20** [comment]
Ok, some more testing, and I think I've got it now:

- Put a raid marker on Lorna so she's easier to see. Move as far away as you can to the edge of her drawing distance. (You can stand south, west, north, ... of her... doesn't matter, the bug happens from every angle.)

- Move back and forth between "she's too far away" and "I can see her"

![pic](http://i.imgur.com/3HKrR5e.jpg)

**The thing is: most of the time she appears properly when you come closer, but in ~ 30% of the cases she does *not* appear as she's supposed to when you move nearer.**
When that happens you can stand directly in front of her without seeing her.

Do not log out now, because that will crash the client when you log in again. To fix this, move back from Lorna until you are very far away, then go back to her until you see her again.

It does not have anything to do with other people as i suspected, because it also happens when I'm alone in Gilneas. It also does not have anything to do with the horses I've mentioned... the horses are just far enough away to be out of her drawing distance. So when gathering the horses you move in and out of her drawing distance a couple of times - every time you have a chance of getting this bug.

**Comment by Tj20** [comment]
(deleted some direct images in previous posts and made them links only)

**Comment by zarevucky** [comment]
Happened to me, too. Started the quest as a cat, when I returned with horses (all five together), she was gone. Decided I do the other quests first, crashed on issue 3329, can't login since.

**Comment by Tj20** [comment]
Today I deleted my cache folder again and after some trying I've noticed that whatever I do, I can't get Lorna to disappear any more, also no more crashes for me on Grandma's Cat (#3329). Even my low level druid which I thought would be permanently stuck now works again, I could just login normally.

Since the server is up since 3.5 hours I assume you have fixed both quests already at some point earlier today. Good job!

**Comment by zarevucky** [comment]
I am still crashing same as before. Deleting cache does nothing. Are you sure you didn't change anything else in the meantime?

**Comment by Tj20** [comment]
Damn... apparently I had spoken too soon :(

When I logged back in today (server uptime: 14 hours), I encountered the same problems:

Lorna kept disappearing: if you go back and forth at the edge of her max rendering distance, sometimes she appears correctly and sometimes not at all.

There was also another guy whose game has crashed apparently still he wasn't moving and continuosly went on/off.

![pic](http://i.imgur.com/IylgtQe.jpg)

I now also believe that the crash at Grandma's Cat (#3329) is closely related to Lorna Crowley.

I've noticed that when Lorna Crowley is visible and I walk from Lorna to the cat I have no problems near the cat. But when I come from the Wahl Cottage my client crashes as soon as it enters the range where Lorna Crowley should be visible to me. That area is near the cat. So I think the cat might be fine and actually Lorna is the problem here.

Here's a screenshot of me spamming /tar Lorna while being near the cat. My client crashes as soon as it enters the area where Lorna should become visible.

![You shall not lay your eye upon me, stranger!](http://i.imgur.com/lrKeDJe.jpg)

I don't know what's wrong with Lorna. Why is she okay 3 hours after a server restart, and 14 hours she's buggy and does sometimes not appear when entering her rendering range?

But I think that this is a big problem. If she is buggy you can sometimes see multiple people standing around her and you know their client crashed. From the 3 guys I saw two days ago 1 deleted his char (not on my friend list any more), and the other 2 apparently have not been played after that.

Seasoned players might not care as much, but new players on our server will probably check out the "new thing" this expansion: Worgen and Goblins. Some will probabably come to the forum and ask for help, but others will probably just leave and never come back.

**Comment by zarevucky** [comment]
I can confirm the opinion about new players. I only came to a Cataclysm server because I wanted to play worgen. Since this is not an option here, I just found myself a WotLK server and playing there.

**Comment by Tj20** [comment]
I believe this issue has been fixed in [rev. 4458](http://atlantiss.eu/changelog#rev4458) by Wolffenstein. 

I tried to reproduce the disappearing Lorna Crowley by going back and forth at the edge of her render distance and couldn'd to it. It also seems that there have been no new complaints about her, so I think this issue has been fixed.
